qiskit_nature/problems/sampling/protein_folding/bead_distances/distance_map_builder.py

```diff
@@ -0,0 +1,427 @@
+
+    distance_map_axis_0[lower_side_bead][upper_side_bead] = _calc_distance_term(
+        peptide, distance_map_axis_0, lower_bead_ind, upper_bead_ind, lower_indic_0, upper_indic_0
+    )
+    distance_map_axis_1[lower_side_bead][upper_side_bead] = _calc_distance_term(
+        peptide,
+        distance_map_axis_1,
+        lower_bead_ind,
+        upper_bead_ind,
+        lower_indic_1,
+        upper_indic_1,
+    )
+    distance_map_axis_2[lower_side_bead][upper_side_bead] = _calc_distance_term(
+        peptide,
+        distance_map_axis_2,
+        lower_bead_ind,
+        upper_bead_ind,
+        lower_indic_2,
+        upper_indic_2,
+    )
+    distance_map_axis_3[lower_side_bead][upper_side_bead] = _calc_distance_term(
+        peptide,
+        distance_map_axis_3,
+        lower_bead_ind,
+        upper_bead_ind,
+        lower_indic_3,
+        upper_indic_3,
+    )
+
+
+def _calc_dists_side_main_all_axes(
+    distance_map_axis_0,
+    distance_map_axis_1,
+    distance_map_axis_2,
+    distance_map_axis_3,
+    lower_bead_ind,
+    indic_funs,
+    lower_side_bead,
+    peptide,
+    upper_bead_ind,
+    upper_main_bead,
+):
+    lower_indic_0, lower_indic_1, lower_indic_2, lower_indic_3 = indic_funs
+
+    distance_map_axis_0[lower_side_bead][upper_main_bead] = _calc_distance_term(
+        peptide, distance_map_axis_0, lower_bead_ind, upper_bead_ind, lower_indic_0, None
+    )
+    distance_map_axis_1[lower_side_bead][upper_main_bead] = _calc_distance_term(
+        peptide, distance_map_axis_1, lower_bead_ind, upper_bead_ind, lower_indic_1, None
+    )
+    distance_map_axis_2[lower_side_bead][upper_main_bead] = _calc_distance_term(
+        peptide, distance_map_axis_2, lower_bead_ind, upper_bead_ind, lower_indic_2, None
+    )
+    distance_map_axis_3[lower_side_bead][upper_main_bead] = _calc_distance_term(
+        peptide, distance_map_axis_3, lower_bead_ind, upper_bead_ind, lower_indic_3, None
+    )
+
+
+def _calc_dists_main_side_all_axes(
+    distance_map_axis_0,
+    distance_map_axis_1,
+    distance_map_axis_2,
+    distance_map_axis_3,
+    indic_funs,
+    lower_bead_ind,
+    lower_bead,
+    peptide,
+    upper_bead_ind,
+    upper_bead,
+):
+    indic_0, indic_1, indic_2, indic_3 = indic_funs
+
+    distance_map_axis_0[lower_bead][upper_bead] = _calc_distance_term(
+        peptide, distance_map_axis_0, lower_bead_ind, upper_bead_ind, None, indic_0
+    )
+    distance_map_axis_1[lower_bead][upper_bead] = _calc_distance_term(
+        peptide, distance_map_axis_1, lower_bead_ind, upper_bead_ind, None, indic_1
+    )
+    distance_map_axis_2[lower_bead][upper_bead] = _calc_distance_term(
+        peptide, distance_map_axis_2, lower_bead_ind, upper_bead_ind, None, indic_2
+    )
+    distance_map_axis_3[lower_bead][upper_bead] = _calc_distance_term(
+        peptide, distance_map_axis_3, lower_bead_ind, upper_bead_ind, None, indic_3
+    )
```

Comment:
In general I feel like this could be written as a `for` loop iterating over the list of `distance_map_axes` and `indic_funcs`. I.e.:
```python
for dist_map_ax, idx in zip(distance_map_axes, indic_funcs):
    ...
```
I assume a similar improvement can be done elsewhere in this file.

Reply:
Great idea.